authority/mgmt/api/handler.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"crypto/x509"
+
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/authority"
+	"github.com/smallstep/certificates/authority/mgmt"
+)
+
+// Handler is the ACME API request handler.
+type Handler struct {
+	db       mgmt.DB
+	auth     *authority.Authority
+	rootPool *x509.CertPool
+}
+
+// NewHandler returns a new Authority Config Handler.
+func NewHandler(auth *authority.Authority) api.RouterHandler {
+	h := &Handler{db: auth.GetAdminDatabase(), auth: auth}
+
+	h.rootPool = x509.NewCertPool()
+	for _, cert := range auth.GetRootX509Certs() {
+		h.rootPool.AddCert(cert)
+	}
+
+	return h
+}
+
+// Route traffic and implement the Router interface.
+func (h *Handler) Route(r api.Router) {
+	authnz := func(next nextHTTP) nextHTTP {
+		return h.extractAuthorizeTokenAdmin(h.requireAPIEnabled(next))
+	}
```

Comment:
Most of the same functionality can be achieved using mTLS. We should consider if we want to change this.

Reply:
Had a convo with @mmalone and he wanted to use tokens because it allowed for proxying. But he wanted to so more to authenticate the body of the request. Like hashing the request in the token or signing over the entire request body using the private key.